### PR TITLE
[4.6] testing: python 3.10 fix

### PR DESCRIPTION
**Pull request by hroncok**
This is a clean cherry-pick of d9ac2efbcdee123f73edf1829c8bd6a91be3b6d2

### CODE REVIEW

**Review by nicoddemus — APPROVED** [review]
Thanks @hroncok!

**Comment by nicoddemus** [comment]
Hmmm the failures need investigation though.

File: testing/test_runner.py [hunk 1]
```diff
@@ -336,10 +336,9 @@ def test_method(self):
         assert reps[2].failed
         assert reps[2].when == "teardown"
         assert reps[2].longrepr.reprcrash.message in (
-            # python3 error
             "TypeError: teardown_method() missing 2 required positional arguments: 'y' and 'z'",
-            # python2 error
-            "TypeError: teardown_method() takes exactly 4 arguments (2 given)",
```
Comment (asottile) [inline testing/test_runner.py:342]: need to keep the python2 error message in this sequence since 4.6.x still supports python 2.x

Comment (hroncok) [reply testing/test_runner.py:342]: on it

**Review by asottile — APPROVED** [review]

**Comment by hroncok** [comment]
Still a lot of CI failures here :/

**Comment by bluetech** [comment]
Unless someone volunteers to fix the 4.6.x CI, I think we can go ahead and merge this, as the failures are unrelated and the patch cannot cause any new ones.

**Comment by nicoddemus** [comment]
> Unless someone volunteers to fix the 4.6.x CI, I think we can go ahead and merge this, as the failures are unrelated and the patch cannot cause any new ones.

I agree, but we will need volunteers to fix CI if we want to make a new 4.6.x release in the future.

**Pull request merged**